locations/spiders/vr_bank.py

```
@@ -0,0 +1,188 @@
+                    except Exception:
+                        pass
+
+                    if open_time and close_time and day:
+                        opening_hours.add_range(
+                            day=DAY_MAPPING[day],
+                            open_time=open_time,
+                            close_time=close_time,
+                            time_format='%H:%M'
+                        )
+                    else:
+                        continue
+
+        return opening_hours.as_opening_hours()
+
+    def parse_details(self, response):
+        if response.status == 403:
+            yield scrapy.Request(
+                url=response.meta.get('url'),
+                callback=self.parse_details,
+                meta={'url': response.meta.get('url')}
+            )
+
+        name = street = zip = city = phone = website = latitude = longitude = ""
+
+        try:
+            name = response.xpath('//h1[@itemprop="name"]/text()').get()
+        except Exception:
+            pass
+
+        try:
+            street = response.xpath('//span[@itemprop="streetAddress"]/text()').get()
+        except Exception:
+            pass
+        try:
+            zip = response.xpath('//span[@itemprop="postalCode"]/text()').get()
+        except Exception:
+            pass
+
+        try:
+            city = response.xpath('//span[@itemprop="addressLocality"]/text()').get()
+        except Exception:
+            pass
+
+        try:
+            phone = response.xpath('//li[@itemprop="telephone"]/a/span/text()').get()
+        except Exception:
+            pass
+
+        try:
+            website = response.xpath('//li[@itemprop="url"]/a/span/text()').get()
+        except Exception:
+            pass
+
+        try:
+            m = re.search(r'lat&quot;:([-+]?[0-9]*\.?[0-9]*)', response.text)
+            if m:
+                latitude = m.group(1)
+        except Exception:
```

Please don't wrap all these in exception catches. If something fails, we want to fix it.

---

Actually these are expected to fail. Like some banks do not provide working hours. In that case the regex will fail.

---

Please add a comment and use a more restrictive exception in the `except` block so we don't swallow all exceptions.

---

Please check now.

---

What I meant by "use a more restrictive exception" is to figure out which exception is thrown that you want to catch and catch only that one instead of catching `Exception`.

---

Done. I have removed unwanted exception handling and elsewhere made it more restrictive to ValueError.